http/server.ts

```diff
@@ -220,70 +221,78 @@ function fixLength(req: ServerRequest): void {
 // ParseHTTPVersion parses a HTTP version string.
 // "HTTP/1.0" returns (1, 0, true).
 // Ported from https://github.com/golang/go/blob/f5c43b9/src/net/http/request.go#L766-L792
-export function parseHTTPVersion(vers: string): [number, number, boolean] {
-  const Big = 1000000; // arbitrary upper bound
-  const digitReg = /^\d+$/; // test if string is only digit
-  let major: number;
-  let minor: number;
-
+export function parseHTTPVersion(vers: string): [number, number] {
   switch (vers) {
     case "HTTP/1.1":
-      return [1, 1, true];
+      return [1, 1];
+
     case "HTTP/1.0":
-      return [1, 0, true];
-  }
+      return [1, 0];
 
-  if (!vers.startsWith("HTTP/")) {
-    return [0, 0, false];
-  }
+    default: {
+      const Big = 1000000; // arbitrary upper bound
```

Comment:
Why is an arbitrary upper bound for the HTTP version necessary here? Isn't HTTP/2 already binary instead of text-based, and doesn't have a status line?

Reply:
See golang api. It's a simple port.

Reply:
It seems out of scope for this PR to fix it, but I agree with @MarkTiedemann, this parser seems rather over-engineered. 
I think we should accept `"HTTP/0.9"` and `/^HTTP\/1.(\d+)$/` and call it a day.

Reply:
SGTM

Reply:
> It seems out of scope for this PR to fix it

Agreed.

> I think we should accept `"HTTP/0.9"` and `/^HTTP\/1.(\d+)$/` and call it a day.

I think we should accept only `0.9`, `1.0`, and `1.1` in this case. This is effectively an HTTP 1 parser. Since H2 and H3 are completely different and there's not gonna be a `1.2` version (at least not any time soon), I think we can simplify this even more. (And even if `1.3` should come out one day, we'll probably know about it in advance and can just simply add the missing case then. I'd go with YAGNI here.)

This brings up another important point (which is unimportant for this PR, however): Do we need to parse and handle version `0.9` and `1.0` differently? For example, [`0.9`](https://www.w3.org/Protocols/HTTP/HTTP2.html) uses different [methods](https://www.w3.org/Protocols/HTTP/Methods.html), such as -- and this is not a joke -- [`SPACEJUMP`](https://www.w3.org/Protocols/HTTP/Methods/SpaceJump.html). Are there important differences that we need to take into account? And on another related note, is anyone still using `0.9` and `1.0`? `1.1` is from 1997, 22 years old. If no one is using `0.9` and `1.0`, should we just drop support for them?

Reply:
Agree with all of you. Created an issue for this: https://github.com/denoland/deno_std/issues/463